docs/book/src/developer/providers/webhooks.md

```diff
@@ -5,6 +5,21 @@ Cluster API provides support for three kinds of webhooks: validating webhooks, d
 ## Validating webhooks
 Validating webhooks are an implementation of a [Kubernetes validating webhook](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#validatingadmissionwebhook). A validating webhook allows developers to test whether values supplied by users are valid. e.g. [the Cluster webhook] ensures the Infrastructure reference supplied at the Cluster's `.spec.infrastructureRef` is in the same namespace as the Cluster itself and rejects the object creation or update if not.
 
+<aside class="note">
+
+<h1> ClusterClass and managed topology support in validating webhooks </h1>
+
+Validating webhooks implemented for a `InfrastructureMachineTemplate` or `BootstrapConfigTemplate` resource
+are required to not block due to immutability checks when the controller for managed
+topology and ClusterClass does [Server Side Apply] dry-run requests.
+
```

Comment:
I would echo ~what we have in the migration doc given that it was well-received by the community 

```suggestion
[Server Side Apply](https://kubernetes.io/docs/reference/using-api/server-side-apply/) implementation in ClusterClass and managed topologies requires to dry-run changes on templates. If infrastructure or bootstrap providers have implemented immutability checks in their InfrastructureMachineTemplate or BootstrapConfigTemplate webhooks, it is required to implement the following changes in order to prevent dry-run to return errors. The implementation requires sigs.k8s.io/controller-runtime in version >= v0.12.3.

In order to do so it is required to use a controller runtime CustomValidator. This will allow to skip the immutability check only when the topology controller is dry running while preserving the validation behavior for all other cases.
```